common/proof_test.go

```diff
@@ -2,14 +2,183 @@ package common_test
 
 import (
 	"errors"
+	"os"
 	"testing"
 
+	"bytes"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"log"
+
 	"github.com/stretchr/testify/require"
 	"github.com/zeta-chain/zetacore/common"
+	"github.com/zeta-chain/zetacore/common/bitcoin"
+	"github.com/zeta-chain/zetacore/x/crosschain/keeper"
+	crosschaintypes "github.com/zeta-chain/zetacore/x/crosschain/types"
+
+	"github.com/btcsuite/btcd/blockchain"
+	"github.com/btcsuite/btcd/btcjson"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/btcsuite/btcutil"
 )
 
+const numBlocksToTest = 100
+
+type Block struct {
+	TssAddress   string `json:"tssAddress"`
+	Height       int    `json:"height"`
+	Nonce        uint64 `json:"nonce"`
+	OutTxid      string `json:"outTxid"`
+	HeaderBase64 string `json:"headerBase64"`
+	BlockBase64  string `json:"blockBase64"`
+}
+
+type Blocks struct {
+	Blocks []Block `json:"blocks"`
+}
+
+func LoadTestBlocks() Blocks {
+	file, err := os.Open("./bitcoin/test_blocks.json")
```

Comment:
We should use `TempDir` for test file
```
	tempDir := os.TempDir()
	file, err := os.Create(filepath.Join(tempDir, "test_blocks.json"))
```

Reply:
test_blocks.json is a piece of test data file (located in `common/test_data/`) in the repo.